### PR TITLE
addded onlyForFemale field

### LINKED ISSUE #61: Add 'onlyForFemale' field in createOpportunity API
In `controllers/opportunity/index.js` , `managers/opportunity/index.js` , `services/opportunity/index.js` , update the `createOpportunity` method to create the Opportunity with one more field `onlyForFemale`.

### PR DESCRIPTION
**Pull request by ritik307**
# Description

Added `onlyForFemale` field in the following:
- [x]  controllers/opportunity/index.js 
- [x]  managers/opportunity/index.js 
- [x]  services/opportunity/index.js

Fixes #61 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ritik307** [comment]
@Manvityagi plz review

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Corresponding additions in the documentation have not been done - 
In `routes/opportunity.js` in the response of `get` and in properties of `post` API documentation. 

Corresponding additions in tests have also not been done. 
Every test should be updated which is affected by this change. 

As a rule of thumb, For a backend PR, all the changes should be accompanied by corresponding changes in tests and documentation.

@ritik307

**Comment by ritik307** [comment]
> 
> 
> Corresponding additions in the documentation have not been done -
> In `routes/opportunity.js` in the response of `get` and in properties of `post` API documentation.
> 
> Corresponding additions in tests have also not been done.
> Every test should be updated which is affected by this change.
> 
> As a rule of thumb, For a backend PR, all the changes should be accompanied by corresponding changes in tests and documentation.
> 
> @ritik307

ok got it will do that ASAP

**Comment by ritik307** [comment]
@Manvityagi  what should I write in the `type` and `description` section of the documentation.

**Comment by Manvityagi** [comment]
> @Manvityagi what should I write in the `type` and `description` section of the documentation.

Represents if the opportunity is only for females

**Comment by ritik307** [comment]
> 
> 
> > @Manvityagi what should I write in the `type` and `description` section of the documentation.
> 
> Represents if the opportunity is only for females

The `type` should be boolean right?

**Comment by Manvityagi** [comment]
> > > @Manvityagi what should I write in the `type` and `description` section of the documentation.
> > 
> > 
> > Represents if the opportunity is only for females
> 
> The` type` should be boolean right?

Yes

**Comment by ritik307** [comment]
@Manvityagi Done with changes Plz check

File: routes/opportunity.js [hunk 1]
```diff
@@ -84,6 +84,10 @@ const opportunityManager = new OpportunityManager(),
  *                         type: string
  *                         description: Link for the opportunity.
  *                         example: www.rohithmsrscholarship.com
+ *                       onlyForFemale:
+ *                         type: bool
```
Comment (Manvityagi) [inline routes/opportunity.js]: It should be `boolean`, check swagger doc

File: routes/opportunity.js [hunk 2]
```diff
@@ -142,6 +147,9 @@ router.get('/', (req, res) => {
  *              opportunityURL:
  *                type: string
  *                description: URL of the opportunity
+ *              onlyForFemale:
+ *                type: bool
```
Comment (Manvityagi) [inline routes/opportunity.js]: Make it `boolean`

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Rest looks good, 

Awaiting @abdus approval also to cross-check if I have missed any part that needed to update but isn't updated.

File: routes/opportunity.js [hunk 3]
```diff
@@ -84,6 +84,10 @@ const opportunityManager = new OpportunityManager(),
  *                         type: string
  *                         description: Link for the opportunity.
  *                         example: www.rohithmsrscholarship.com
+ *                       onlyForFemale:
+ *                         type: bool
+ *                         description: Represents if the opportunity is only for females   
+ *                         example: Female Only: true
```
Comment (Manvityagi) [inline routes/opportunity.js]: This should be only `example: true`

**Review by Manvityagi — CHANGES_REQUESTED** [review]

File: routes/opportunity.js (hunk 3 above)
Comment (ritik307) [reply routes/opportunity.js]: Done!

File: routes/opportunity.js (hunk 2 above)
Comment (ritik307) [reply routes/opportunity.js]: Done!

File: routes/opportunity.js (hunk 1 above)
Comment (ritik307) [reply routes/opportunity.js]: Done!

**Comment by ritik307** [comment]
@Manvityagi I have made all the requested changes plz review

**Review by Manvityagi — APPROVED** [review]

**Comment by ritik307** [comment]
@Manvityagi THnaks for merging the PR Kindly add the appropriate GSSOC21 tag and level tag to the PR.

**Pull request merged**